### PR TITLE
🔀 merge updates from source repo

### PR DESCRIPTION
**Pull request by migbro**
# Pull Request Template

## Description

GATK4 workflow has been refactored to more strictly match the Broad WDL, and to mirror the COV-IRT HEC pipeline, with the exception of genotype-calling specific params. It also incorporates knowledge from the COV-IRT pipeline group, namely Robert Meller in which BQSR should be run on the input bam as a whole, and not split by intervals.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Extensive testing in the Cavatica environment, with output comaprisons on various file splitting conditions show that, without extensive benchmarking, the differing result obtained from haplotype caller when splitting input bams for processing versus not (except for haplotype calling itself) does not produce exact results and can't be confirmed to be "good enough." This pipeline is slower that the previous but adheres to Broad run style.

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: subworkflows/preprocess_rnaseq_bam.cwl [hunk 1]
```diff
@@ -1,7 +1,8 @@
 cwlVersion: v1.0
 class: Workflow
 doc: >-
-    "Sambamba Sort Mark Dup Sub WF"
+    "Preprocessing sub workflow for GATK4 RNAseq SNV calling.
+    Uses Sambamba for mark dups and sort, GATK SplitNCigarReads for final processed bam output"
```
Comment (kmhernan) [inline subworkflows/preprocess_rnaseq_bam.cwl:5]: While sambamba is much more efficient than Picard, the implementations are not exactly the same. I'm not so concerned about it, but it's important to be aware of this.

Comment (migbro) [reply subworkflows/preprocess_rnaseq_bam.cwl:5]: Thanks! I've made more updates in which I use sambamba just once - no intermediate sort steps. As for picard compatibility, I can do some benchmarking, but am also ok with general consensus that the effect is minimal.

Comment (migbro) [reply subworkflows/preprocess_rnaseq_bam.cwl:5]: Added a qualifying sentence to the readme. A fair amount of updates from the last PR...I could try and close this one and instead do it by release tag to that PR #1 goes to v0.3, PR # v0.4

**Review by kmhernan — APPROVED** [review]
LGTM 🙏🏽

**Pull request merged**